src/xb-machine.c

```diff
@@ -1399,53 +1365,44 @@ xb_machine_func_eq_cb (XbMachine *self,
 					xb_opcode_cmp_val_or_str,
 					error))
 		return FALSE;
-
-	xb_machine_stack_pop (self, stack, &op1);
-	xb_machine_stack_pop (self, stack, &op2);
+	if (!xb_machine_stack_pop (self, stack, &op1, error))
+		return FALSE;
+	if (!xb_machine_stack_pop (self, stack, &op2, error))
+		return FALSE;
```

Comment:
I think various other bits of the code attempt to leave the stack unchanged if they error out, so it would probably make sense to add a helper function for popping two opcodes off the stack at once, doing all the stack size checks only once, and making sure that the stack is unchanged if an error is returned.

Reply:
Yes, probably a nice optimisation for the common case too. Will fix.